master/buildbot/process/buildstep.py

```diff
@@ -658,12 +658,25 @@ def start(self):
         raise NotImplementedError("your subclass must implement run()")
 
     def interrupt(self, reason):
+        # TODO: consider adding an INTERRUPTED or STOPPED status to use
```

Comment:
I would appreciate you remove those 3 lines, and make sure that CANCELLED is set as the status if self.stopped is True